src/platform/nxp/common/NetworkCommissioningWiFiDriver.cpp

```diff
@@ -291,8 +291,24 @@ CHIP_ERROR NXPWiFiDriver::StartScanWiFiNetworks(ByteSpan ssid)
     return CHIP_NO_ERROR;
 }
 
-// TODO should be modified to do it in the context of the Matter stack
-int NXPWiFiDriver::OnScanWiFiNetworkDone(unsigned int count)
+int NXPWiFiDriver::_OnScanWiFiNetworkDoneCallBack(unsigned int count)
+{
+    int error_code = WM_SUCCESS;
+    ChipDeviceEvent event;
+    event.Type                          = DeviceEventType::kPlatformNxpScanWiFiNetworkDoneEvent;
+    event.Platform.ScanWiFiNetworkCount = count;
+    CHIP_ERROR err                      = PlatformMgr().PostEvent(&event);
+    if (err != CHIP_NO_ERROR)
+    {
+        ChipLogError(DeviceLayer, "Failed to schedule work: %" CHIP_ERROR_FORMAT, err.Format());
+        error_code = WM_FAIL;
+    }
+    return error_code;
```

Comment:
This style is not ideal for maintainability: for me to see what error_code is returned (on line 306) I need to scan through the entire function and find the initialization at line 296

```suggestion
    ChipDeviceEvent event;
    event.Type                          = DeviceEventType::kPlatformNxpScanWiFiNetworkDoneEvent;
    event.Platform.ScanWiFiNetworkCount = count;
    CHIP_ERROR err                      = PlatformMgr().PostEvent(&event);
    if (err != CHIP_NO_ERROR)
    {
        ChipLogError(DeviceLayer, "Failed to schedule work: %" CHIP_ERROR_FORMAT, err.Format());
        return WM_FAIL;
    }
    return WM_SUCCESS;
```

Reply:
Thank you for your suggestion, I did the update